docs/reverse_proxy.md

```diff
@@ -102,6 +103,7 @@ example.com:8448 {
     SSLEngine on
     ServerName matrix.example.com;
 
+    RequestHeader set "X-Forwarded-Proto" expr=%{REQUEST_SCHEME}
```

Comment:
Apache and caddy both set X-Forwarded-For by default, though in a way that https://github.com/matrix-org/synapse/issues/9471 is problematic.